elements/lisk-chain/src/diff.ts

```diff
@@ -130,15 +130,104 @@ const diffAlgo = (initial: Buffer, final: Buffer): HistoryType[] => {
 	return [];
 };
 
+/**
+ * This function returns the length of common prefix between two buffers
+ */
+const diffCommonPrefix = (buffer1: Buffer, buffer2: Buffer): number => {
+	// Quick check for common null cases.
+	if (buffer1[0] !== buffer2[0]) {
+		return 0;
+	}
+	// Binary search.
+	// Performance analysis: http://neil.fraser.name/news/2007/10/09/
+	let pointerMin = 0;
+	let pointerMax = Math.min(buffer1.length, buffer2.length);
+	let pointerMid = pointerMax;
+	let pointerstart = 0;
+
+	while (pointerMin < pointerMid) {
+		if (
+			buffer1
+				.slice(pointerstart, pointerMid)
+				.equals(buffer2.slice(pointerstart, pointerMid))
+		) {
+			pointerMin = pointerMid;
+			pointerstart = pointerMin;
+		} else {
+			pointerMax = pointerMid;
+		}
+		pointerMid = Math.floor((pointerMax - pointerMin) / 2 + pointerMin);
+	}
+
+	return pointerMid;
+};
+
+/**
+ * This function returns the length of common suffix between two buffers
+ */
+const diffCommonSuffix = (buffer1: Buffer, buffer2: Buffer): number => {
+	// Quick check for common null cases.
+	if (!buffer1.slice(-1).equals(buffer2.slice(-1))) {
+		return 0;
+	}
+	// Binary search.
+	// Performance analysis: http://neil.fraser.name/news/2007/10/09/
+	let pointerMin = 0;
+	let pointerMax = Math.min(buffer1.length, buffer2.length);
+	let pointerMid = pointerMax;
+	let pointerEnd = 0;
+
+	while (pointerMin < pointerMid) {
+		if (
+			buffer1
```

Comment:
might be good to use compare?
https://nodejs.org/api/buffer.html#buffer_buf_compare_target_targetstart_targetend_sourcestart_sourceend
i didnt check performance tho

Reply:
It slightly degrades the performance 😕